python-mini-projects/projects/convert pdf to text/converter1.py

```diff
@@ -0,0 +1,43 @@
+# -*- coding: utf-8 -*-
+
+
+import PyPDF2
+import os
+
+
+if(os.path.isdir("temp") == False):
+    os.mkdir("temp")
+    
+txtpath = ""
+pdfpath = ""
+
+
+
+pdfpath = input("Enter the name of your pdf file - please use backslash when typing in directory path: ")   #Provide the path for your pdf here
+txtpath = input("Enter the name of your txt file - please use backslash when typing in directory path: ")   #Provide the path for the output text file  
+
+BASEDIR = os.path.realpath("temp") # This is the sample base directory where all your text files will be stored if you do not give a specific path
+print(BASEDIR)
+
+
+if(len(txtpath) == 0):
+    txtpath = os.path.join(BASEDIR,os.path.basename(os.path.normpath(pdfpath)).replace(".pdf", "")+".txt")
+pdfobj = open(pdfpath, 'rb')
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the following resource: *pdfobj*. The resource is allocated by call *builtins.open*. The resource is referenced at lines 27, 41. The resource is closed at line 41. There are other execution paths that do not contain closure statements, for example, when *builtins.print()* throws an exception. To prevent this resource leak, close *pdfobj* in a try-finally block or declare it using a `with` statement.

**More info**
[View details about the `with` statement in the Python developer&#39;s guide](https://www.python.org/dev/peps/pep-0343/) (external link).